### DISCUSSION TITLE
The value of "service.instance.id" changes every time the server restarts. Is it okay to make it fixed?

### DISCUSSION THREAD

**Discussion by zerocooldog** [open-telemetry/opentelemetry-java-instrumentation#12312] [Q&A]
We assign and manage a value called agent id for each WAS (Web Application Server) configuration. Once an agent id is generated, we reuse the initially created value for the same WAS configuration instead of generating a new one.

It seems that the service.instance.id acts as an identifier, similar to the agent id we use, but we've noticed that its value changes every time the server restarts.

In our implementation of ResourceProvider, we're considering using a fixed agent ID that persists across server restarts. Could you please clarify if there are any implications or drawbacks to this approach?

```
public class Hamonica2ResourceProvider implements ResourceProvider {

    @Override
    public Resource createResource(ConfigProperties configProperties) {

        Attributes attributes = Attributes.builder().put("service.instance.id", "test").build();
        return Resource.create(attributes);
    }
}

```

Also, we would like to know how to remove the following attributes:

```
attributes {
  key: "telemetry.distro.name"
  value {
    string_value: "opentelemetry-java-instrumentation"
  }
}
attributes {
  key: "telemetry.distro.version"
  value {
    string_value: "demo-2.0.0-SNAPSHOT-otel-2.9.0-SNAPSHOT"
  }
}
```

**Comment by laurit** [comment]
You can read about `service.instance.id` at https://opentelemetry.io/docs/specs/semconv/attributes-registry/service/#service-attributes Questions about semantic conventions are best asked in the semantic conventions repository https://github.com/open-telemetry/semantic-conventions/issues
